packages/manager/cypress/component/features/firewalls/firewall-rule-table.spec.tsx

```diff
@@ -0,0 +1,154 @@
+  '22': 'SSH',
+  '53': 'DNS',
+  '80': 'HTTP',
+  '443': 'HTTPS',
+  '3306': 'MySQL',
+};
+
+const mockInboundRule = (label: string) => {
+  return firewallRuleFactory.build({
+    action: 'ACCEPT',
+    description: randomString(),
+    label,
+    ports: randomItem(Object.keys(portPresetMap)),
+  });
+};
+
+const mockOutboundRule = (label: string) => {
+  return firewallRuleFactory.build({
+    action: 'DROP',
+    description: randomString(),
+    label,
+    ports: randomItem(Object.keys(portPresetMap)),
+  });
+};
+
+componentTests('Firewall Rules', (mount) => {
+  describe('firewall inbound rules drag & drop with mouse', () => {
+    const inboundRule1: FirewallRuleType = mockInboundRule('inbound_rule_1');
+    const inboundRule2: FirewallRuleType = mockInboundRule('inbound_rule_2');
+    const inboundRule3: FirewallRuleType = mockInboundRule('inbound_rule_3');
+
+    // Note that nth-child is 1-indexed
+    const tableRow = 'div[aria-label="inbound Rules List"] tbody tr';
+    const firstRow =
+      'div[aria-label="inbound Rules List"] tbody tr:nth-child(1)';
+    const secondRow =
+      'div[aria-label="inbound Rules List"] tbody tr:nth-child(2)';
+
+    beforeEach(() => {
+      // Set the viewport to 1024x768
+      cy.viewport(1024, 768);
+
+      // Mount the FirewallRulesLanding
+      mount(
+        <FirewallRulesLanding
+          rules={{
+            inbound: [inboundRule1, inboundRule2, inboundRule3],
+            inbound_policy: 'ACCEPT',
+            outbound_policy: 'DROP',
+          }}
+          disabled={false}
+          firewallID={randomNumber()}
+        />
+      );
+    });
+
+    // Test the drag and drop functionality using mouse
+    it('should move inbound rules rows using mouse drag and drop', () => {
+      // Drag the 1st row rule to 2nd row position
+      // Note that eq is 0-indexed
+      cy.get(tableRow).eq(0).drag(secondRow);
+
+      // Verify the labels in the 1st, 2nd, and 3rd rows
+      cy.get(tableRow).eq(0).should('contain', inboundRule2.label);
+      cy.get(tableRow).eq(1).should('contain', inboundRule1.label);
+      cy.get(tableRow).eq(2).should('contain', inboundRule3.label);
+
+      // Drag the 3rd row rule to 2nd row position
+      cy.get(tableRow).eq(2).drag(secondRow);
+
+      // Verify the labels in the 1st, 2nd, and 3rd rows
+      cy.get(tableRow).eq(0).should('contain', inboundRule2.label);
+      cy.get(tableRow).eq(1).should('contain', inboundRule3.label);
+      cy.get(tableRow).eq(2).should('contain', inboundRule1.label);
+
+      // Drag the 3rd row rule to 1st position
+      cy.get(tableRow).eq(2).drag(firstRow);
+
+      // Verify the labels in the 1st, 2nd, and 3rd rows
+      cy.get(tableRow).eq(0).should('contain', inboundRule1.label);
+      cy.get(tableRow).eq(1).should('contain', inboundRule2.label);
+      cy.get(tableRow).eq(2).should('contain', inboundRule3.label);
+    });
+  });
+
+  describe('firewall outbound rules drag & drop with mouse', () => {
```

Comment:
```suggestion
  describe('should move outbound rules rows using mouse drag and drop', () => {
```

nit: we could make the description format consistent

Reply:
name is consistent for both describe block, 'should' should be used with 'it'
'firewall inbound rules drag & drop with mouse'
'firewall outbound rules drag & drop with mouse'

Reply:
yea, My bad. Just saw it